### PR TITLE
feat: Add node's ssh key to Juju model on join

### PR DESCRIPTION
**Pull request by wyattrees**
Uses `juju add-ssh-key` to add the node's public key to the model, allowing users to ssh to each machine with `juju ssh`

### CODE REVIEW

File: anvil-python/anvil/utils.py
Comment (skatsaounis) [inline anvil-python/anvil/utils.py:1]: Are the changes in this file really needed or are they leftovers from development?

File: anvil-python/anvil/commands/juju.py
Comment (skatsaounis) [inline anvil-python/anvil/commands/juju.py:24]: Nit:

```suggestion
class JujuAddSSHKeyStep(BaseStep):
```

If you agree with the change you also need to change it to `provider/local/commands.py`; import+usage

File: anvil-python/anvil/provider/local/commands.py
Comment (skatsaounis) [inline anvil-python/anvil/provider/local/commands.py:468]: IMHO this has to be the last step of `plan1` because this is the place where you are adding the machine to juju model.

**Review by skatsaounis — CHANGES_REQUESTED** [review]
Overall the changes look pretty good. I added some comments which can easily be resolved and after that we can merge the PR :slightly_smiling_face:

File: anvil-python/anvil/utils.py
Comment (wyattrees) [reply anvil-python/anvil/utils.py:1]: not needed. I (incorrectly) assumed our formatter would remove these imports

**Review by skatsaounis — APPROVED** [review]
LGTM +1

**Comment by skatsaounis** [comment]
@wyattrees could you please sign the commits in order to merge this? Feel free to also squash into 1 with only you as a committer. :slightly_smiling_face:

**Pull request merged**
